toolchain/check/import.cpp

```diff
@@ -273,11 +273,41 @@ static auto AddImportRefOrMerge(Context& context, SemIR::ImportIRId ir_id,
   auto& entry = parent_scope.GetEntry(entry_id);
   if (inserted) {
     auto entity_name_id = context.entity_names().Add(
-        {.name_id = name_id, .parent_scope_id = parent_scope_id});
-    entry.result = SemIR::ScopeLookupResult::MakeFound(
-        AddImportRef(context, {.ir_id = ir_id, .inst_id = import_inst_id},
-                     entity_name_id),
-        SemIR::AccessKind::Public);
+      {.name_id = name_id, .parent_scope_id = parent_scope_id});
+    auto import_ref = AddImportRef(context, {.ir_id = ir_id, .inst_id = import_inst_id},
+      entity_name_id);
+    entry.result = SemIR::ScopeLookupResult::MakeFound( import_ref,
+      SemIR::AccessKind::Public);
+
+    // Import references for non-owning declarations that match current library
+    const auto functions = import_sem_ir.get_non_owning_functions();
+    auto current_id = context.sem_ir().library_id();
+    bool is_current_default = current_id == SemIR::LibraryNameId::Default;
+    for (auto function_id : functions) {
+      const auto& function = import_sem_ir.functions().Get(function_id);
+      auto lib_id = function.extern_library_id;
+      bool is_lib_default = lib_id == SemIR::LibraryNameId::Default;
+
+      if ((is_lib_default && is_current_default) ||
+          (!is_lib_default && !is_current_default)) {
+        if (is_lib_default) {  // both libraries are default, import ref.
+          LoadImportRef(context, import_ref);
+        } else {
+          // both libraries are non-default, check if they're the same named
```

Comment:
```suggestion
          // Both libraries are non-default: check if they're the same named
```

Style nits: capitalization